examples/favorites_advanced/favorites_advanced_base/lib/src/theme/hotel_app_theme.dart

```diff
@@ -1,6 +1,71 @@
+import 'package:favorites_advanced_base/core.dart';
 import 'package:flutter/material.dart';
 
 class HotelAppTheme {
+  static ThemeData buildTheme(DesignSystem designSystem) {
+    final designSystemColor = designSystem.colors;
+    final colorSchemeTheme = designSystemColor.brightness == Brightness.light
+        ? ColorScheme.light()
+        : ColorScheme.dark();
+    final ColorScheme colorScheme = colorSchemeTheme.copyWith(
+      primary: designSystemColor.primaryColor,
+      secondary: designSystemColor.secondaryColor,
+    );
+
+    final ThemeData base = designSystemColor.brightness == Brightness.light
+        ? ThemeData.light()
+        : ThemeData.dark();
+
+    return base.copyWith(
+      colorScheme: colorScheme,
+      primaryColor: designSystemColor.primaryColor,
+      buttonColor: designSystemColor.buttonColor,
+      indicatorColor: designSystemColor.indicatorColor,
+      splashColor: designSystemColor.splashColor,
+      splashFactory: InkRipple.splashFactory,
+      accentColor: designSystemColor.accentColor,
+      canvasColor: designSystemColor.canvasColor,
+      backgroundColor: designSystemColor.backgroundColor,
+      scaffoldBackgroundColor: designSystemColor.scaffoldBackgroundColor,
+      errorColor: designSystemColor.errorColor,
+      buttonTheme: ButtonThemeData(
+        colorScheme: colorScheme,
+        textTheme: ButtonTextTheme.primary,
+      ),
+      textTheme: _buildDesignTextTheme(base.textTheme, designSystemColor),
+      primaryTextTheme:
+          _buildDesignTextTheme(base.primaryTextTheme, designSystemColor),
+      accentTextTheme:
+          _buildDesignTextTheme(base.accentTextTheme, designSystemColor),
+      appBarTheme: AppBarTheme(
+        color: designSystemColor.primaryVariant,
+      ),
+      platform: TargetPlatform.iOS,
+    );
+  }
+
+  static TextTheme _buildDesignTextTheme(
+      TextTheme base, DesignSystemColor designSystemColor) {
+    const String fontName = 'WorkSans';
+    return base.copyWith(
+      headline1: base.headline1!.copyWith(fontFamily: fontName),
+      headline2: base.headline2!.copyWith(fontFamily: fontName),
+      headline3: base.headline3!.copyWith(fontFamily: fontName),
+      headline4: base.headline4!.copyWith(fontFamily: fontName),
+      headline5: base.headline5!.copyWith(fontFamily: fontName),
+      headline6: base.headline6!.copyWith(fontFamily: fontName),
+      button: base.button!.copyWith(fontFamily: fontName),
+      caption: base.caption!.copyWith(fontFamily: fontName),
+      bodyText1: base.bodyText1!.copyWith(
+          fontFamily: fontName, color: designSystemColor.bodyTextColor1),
+      bodyText2: base.bodyText2!.copyWith(
+          fontFamily: fontName, color: designSystemColor.bodyTextColor2),
+      subtitle1: base.subtitle1!.copyWith(fontFamily: fontName),
+      subtitle2: base.subtitle2!.copyWith(fontFamily: fontName),
+      overline: base.overline!.copyWith(fontFamily: fontName),
+    );
+  }
+
```

Comment:
@nikolayIvanovPrime, do we use the method `buildLightTheme()` ? If not, can you please remove it to avoid any confusion

Reply:
Not quite sure,  should double check that. But since it's in the core folder I thought It's supposed to be used in the other's samples we have, but because we've implemented the design system only for the booking_app sample, if we remove it the others samples may crash.